File: amdegregorio.geometrycalc/src/amdegregorio/rectangle.clj
(ns amdegregorio.rectangle)

(defprotocol Geo 
  (calculateArea [this] "The area")
  (calculatePerimeter [this] "The perimeter"))

(defrecord Rectangle [length width] 
  Geo
  (calculateArea [this] (* length width))
  (calculatePerimeter [this] (* 2 (+ length width))))

File: GeometryCalcKotlin/src/amdegregorio/geo/Rectangle.kt
package amdegregorio.geo

data class Rectangle(val length: Int, val width: Int) {
	constructor() : this(0,0)
	
	fun calculateArea(): Int {
		return length * width;
	}
	
	fun calculatePerimeter(): Int {
		return 2 * (length + width);
	}
}